circleguard/circleguard.py

```diff
@@ -56,8 +56,11 @@ def __init__(self, key, db_path=None, slider_dir=None, loader=None, cache=True):
         self.loader = Loader(key, cacher=self.cacher) if loader is None else loader(key, self.cacher)
         if slider_dir is None:
             # have to keep a reference to it or the folder gets deleted and can't be walked by Library
-            self.__slider_dir = TemporaryDirectory()
-            self.library = Library(self.__slider_dir.name)
+            self.slider_dir = TemporaryDirectory()
+            # create db and immediately disconnect
+            tmp_library = Library.create_db(self.slider_dir.name)
```

Comment:
There's no advantage to using `create_db` here because there are no pre-existing osu files. We don't need to make a library at all here; rely on 

```python
if self.library is None:
    # connect to library since it's a temporary one
    library = Library(self.slider_dir.name)
```

later on to create the db the first time.